### PR TITLE
Make modules independent of each other and add debug overlay file.

### PR DESCRIPTION
**Pull request by simensrostad**
Its now possible to configure modules out without build errors. This PR includes:
 - Adjustments to module/dependency Kconfigs and source code.
 - Removal of debug log configs in prj.confs and introduce a debug overlay file.
 - Added missing ```k_delayed_work_init()``` function that caused hard fault in the system work queue when a module module that
  was configured out was not responding. (This goes for unresponsive modules in general).

### CODE REVIEW

File: prj_nrf9160dk_nrf9160ns.conf [hunk 1]
```diff
@@ -9,68 +9,48 @@ CONFIG_NEWLIB_LIBC=y
...
 
+# BSD library
+CONFIG_BSD_LIBRARY=y
+
 # Network
 CONFIG_NETWORKING=y
 CONFIG_NET_SOCKETS=y
 CONFIG_NET_SOCKETS_OFFLOAD=y
+## Disable native network stack to save some memory.
+CONFIG_NET_IPV4=n
```
Comment (jtguggedal) [inline prj_nrf9160dk_nrf9160ns.conf]: I think these lines should be replaced with `CONFIG_NET_NATIVE=n`

File: src/ext_sensors/Kconfig [hunk 2]
```diff
@@ -11,9 +11,11 @@ if EXTERNAL_SENSORS
 
 config ACCELEROMETER_DEV_NAME
 	string "Accelerometer device name"
+	default "ADXL362"
```
Comment (jtguggedal) [inline src/ext_sensors/Kconfig]: `default "ADXL362" if BOARD_THINGY91_NRF9160NS`

File: src/ext_sensors/Kconfig [hunk 3]
```diff
@@ -11,9 +11,11 @@ if EXTERNAL_SENSORS
...
 
 config MULTISENSOR_DEV_NAME
 	string "Multisensor device name"
+	default "BME680"
```
Comment (jtguggedal) [inline src/ext_sensors/Kconfig]: `default "ADXL362" if BOARD_THINGY91_NRF9160NS`

File: src/cloud/cloud_codec/cloud_codec.h [hunk 4]
```diff
@@ -190,29 +190,35 @@ int cloud_codec_encode_batch_data(
 void cloud_codec_populate_sensor_buffer(
 				struct cloud_data_sensors *sensor_buffer,
 				struct cloud_data_sensors *new_sensor_data,
-				int *head_sensor_buf);
+				int *head_sensor_buf,
+				size_t buffer_count);
```
Comment (jtguggedal) [inline src/cloud/cloud_codec/cloud_codec.h:194]: Is it intentional to include the buffer changes in the commit named `config: Make module independent and remove debug from prj.conf`?

Comment (simensrostad) [reply src/cloud/cloud_codec/cloud_codec.h:194]: Yes. The commit moves the ```BUFFER_MAX``` configs from ```cloud_codec/Kconfig``` to ```Kconfig.data_module```. This was done to limit dependencies between Kconfigs in the source code. ```ringbuffer.c``` was using them to iterate over the buffers when it seemed cleaner to just pass in the size and only use the ```BUFFER_MAX``` configs int the data module when initially declaring the buffers.

**Review by jtguggedal — APPROVED** [review]

**Pull request merged**